cartographer_ros/cartographer_ros/assets_writer.cc

```diff
@@ -21,12 +21,37 @@
 #include "cartographer/io/ply_writing_points_processor.h"
 #include "cartographer/io/points_processor.h"
 #include "cartographer/io/xray_points_processor.h"
+#include "cartographer_ros/map_writer.h"
+#include "cartographer_ros/occupancy_grid.h"
+#include "nav_msgs/OccupancyGrid.h"
 
 namespace cartographer_ros {
 
 namespace carto = ::cartographer;
 
-void WriteAssets(const std::vector<::cartographer::mapping::TrajectoryNode>&
+void WriteCommonAssets(const std::vector<::cartographer::mapping::TrajectoryNode>&
+                       trajectory_nodes, const std::string& stem) {
+  // Write the trajectory.
+  std::ofstream proto_file(stem + ".pb",
+                           std::ios_base::out | std::ios_base::binary);
+  const carto::proto::Trajectory trajectory =
+      carto::mapping::ToProto(trajectory_nodes);
+  CHECK(trajectory.SerializeToOstream(&proto_file)) << "Could not write trajectory.";
```

Comment:
Also call proto_file.close() and CHECK(proto_file) afterwards. Otherwise if writing fails late this could fail silently.